components/List/src/MDCSelfSizingStereoCell.m

```diff
@@ -0,0 +1,464 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCSelfSizingStereoCell.h"
+
+#import <MDFInternationalization/MDFInternationalization.h>
+#import "MaterialInk.h"
+#import "MaterialTypography.h"
+
+static const CGFloat kVerticalMarginMin = 8.0;
+static const CGFloat kVerticalMarginMax = 16.0;
+static const CGFloat kHorizontalMargin = 16.0;
+
+static const CGFloat kImageSideLengthMedium = 40.0;
+static const CGFloat kImageSideLengthMax = 56.0;
+
+static const CGFloat kTitleColorOpacity = 0.87f;
+static const CGFloat kDetailColorOpacity = 0.6f;
+
+@interface MDCSelfSizingStereoCellLayout : NSObject
+
+@property (nonatomic, assign, readonly) CGFloat cellWidth;
+@property (nonatomic, assign, readonly) CGFloat calculatedHeight;
+@property (nonatomic, assign, readonly) CGRect textContainerFrame;
+@property (nonatomic, assign, readonly) CGRect titleLabelFrame;
+@property (nonatomic, assign, readonly) CGRect detailLabelFrame;
+@property (nonatomic, assign, readonly) CGRect leadingImageViewFrame;
+@property (nonatomic, assign, readonly) CGRect trailingImageViewFrame;
+
+- (instancetype)initWithLeadingImageView:(UIImageView *)leadingImageView
+                       trailingImageView:(UIImageView *)trailingImageView
+                           textContainer:(UIView *)textContainer
+                              titleLabel:(UILabel *)titleLabel
+                             detailLabel:(UILabel *)detailLabel
+                               cellWidth:(CGFloat)cellWidth;
+
+@end
+
+@interface MDCSelfSizingStereoCellLayout ()
+
+@property (nonatomic, assign) CGFloat cellWidth;
+@property (nonatomic, assign) CGFloat calculatedHeight;
+@property (nonatomic, assign) CGRect textContainerFrame;
+@property (nonatomic, assign) CGRect titleLabelFrame;
+@property (nonatomic, assign) CGRect detailLabelFrame;
+@property (nonatomic, assign) CGRect leadingImageViewFrame;
+@property (nonatomic, assign) CGRect trailingImageViewFrame;
+
+@end
+
+@implementation MDCSelfSizingStereoCellLayout
```

Comment:
Consider breaking this out to a separate file in `private/`

Reply:
Ah, yep, I meant to do that! Thanks

Reply:
Done in 44358f1de7ac3e3070e441c76daa13b240dec8ff